### PR TITLE
Fix warnings due to logm in tests/linalg3.jl

### PR DESCRIPTION
**Pull request by mfasi**
Fixing the issue originated in JuliaLang/julia#12217

### CODE REVIEW

File: test/linalg3.jl [hunk 1]
```diff
@@ -161,7 +161,10 @@ for elty in (Float32, Float64, Complex64, Complex128)
                                               0  -0.000000000000002   3.000000000000000])
 end
 
+OLD_STDERR = STDERR
+redirect_stderr(open("/dev/null", "w"))
```
Comment (yuyichao) [inline test/linalg3.jl]: Can you do this one Windows?

**Comment by jiahao** [comment]
Why is the test failing? Suppressing stderr is generally not a good idea.

**Comment by mfasi** [comment]
It's not failing, it's a warning that I expect (line coverage).

File: test/linalg3.jl (hunk 1 above)
Comment (mfasi) [inline test/linalg3.jl]: Good point, I hand't thought about it. I guess it doesn't, I have to find another solution. Thank you for pointing this out.

Comment (yuyichao) [inline test/linalg3.jl]: I think you can do [this](https://github.com/JuliaLang/julia/blob/8566c4c5b1ef3c3525fd526888f0db9a6df3ce26/test/abstractarray.jl#L452)

Comment (yuyichao) [inline test/linalg3.jl]: And you can probably even check if the warning contains what you expect.

File: test/linalg3.jl [hunk 2]
```diff
@@ -161,7 +161,10 @@ for elty in (Float32, Float64, Complex64, Complex128)
                                               0  -0.000000000000002   3.000000000000000])
 end
 
+OLD_STDERR = STDERR
+redirect_stderr()
 for elty in (Float64, Complex{Float64})
+
```
Comment (tkelman) [inline test/linalg3.jl]: delete this line, it fails `make check-whitespace`

Comment (mfasi) [inline test/linalg3.jl]: @tkelman Thank you, I submitted a new version. It still fails on AppVeyor, but the problem does not seem related to the code I modified (there's no error at all, actually).

File: test/linalg3.jl (hunk 1 above)
Comment (mfasi) [inline test/linalg3.jl]: @yuyichao Thank you for your help, I did as you suggested. I check the content of the warning as well.

File: test/linalg3.jl (hunk 2 above)
Comment (yuyichao) [inline test/linalg3.jl]: Did AppVeyor timeout before (exactly) 40 mins?

Comment (mfasi) [inline test/linalg3.jl]: Yes, the second build (ARCH=x86_64, JULIA_CPU_CORES=1) timed out after 39:55. The last lines it printed are
```
c:\projects\julia\base\precompile.jl
 g++ -march=x86-64 -m64  -shared  -L/c/projects/julia/usr/lib/julia -L/c/projects/julia/usr/lib -L/c/projects/julia/usr/bin -o /c/projects/julia/usr/lib/julia/sys.dll /c/projects/julia/usr/lib/julia/sys.o -ljulia $([ WINNT = WINNT ] && echo '' -lssp)
make[1]: Leaving directory `/c/projects/julia'
make[1]: Entering directory `/c/projects/julia/test'
 gcc -march=x86-64 -m64  -O0 -ggdb3 -DJL_DEBUG_BUILD -fstack-protector-all -O3 ccalltest.c  -shared -o libccalltest.dll  -DCC="gcc -march=x86-64 -m64"
make[1]: Leaving directory `/c/projects/julia/test'
usr\bin\julia -e "versioninfo()"
```

**Comment by amitmurthy** [comment]
Same here - https://ci.appveyor.com/project/StefanKarpinski/julia/build/1.0.7063/job/s9wi8a78aqu6ld2t

**Comment by tkelman** [comment]
Timeout is fixed now, so this looks good. Any news on the license issue? If not, we can add the entire license text for the original code and modify the header for the file in which it occurs.

**Comment by mfasi** [comment]
@tkelman Thank you. No news regarding the license yet, but I'm confident I'll have some by Friday, so I'd suggest not to add anything now.

**Comment by mfasi** [comment]
@tkelman I just received a reply that settled the licensing issue. Also, I amended my first commit removing the test causing #12294. This does not fix the problem, it's just to let other people run the tests without unnecessary failures while I continue debugging.

**Comment by mfasi** [comment]
@tkelman Thank you, I tried but my email was rejected. It looks like I do not have permission to post messages to julia-admin. Should I try something else?

File: base/linalg/triangular.jl [hunk 3]
```diff
@@ -1075,6 +1078,7 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
     return UpperTriangular(Y)
 
 end
+logm(A::LowerTriangular) = logm(A.').'
```
Comment (tkelman) [inline base/linalg/triangular.jl]: is this also correct for complex element type?

Comment (mfasi) [inline base/linalg/triangular.jl]: I think it is, I just use the fact that if you transpose an `UpperTriangular` you get a `LowerTriangular`.

**Comment by tkelman** [comment]
Maybe forward the email directly to @StefanKarpinski and @ViralBShah just so they can have a written record?

**Comment by mfasi** [comment]
@tkelman I did as you suggested, thank you.

**Comment by ViralBShah** [comment]
@mfasi Thanks for the email.

**Pull request merged**